src/app/api/edit-profile/route.client.ts

```diff
@@ -0,0 +1,56 @@
+import { z } from "zod";
+import { EditProfileRequest, EditProfileResponse } from "./route.schema";
+
+type IEditProfileRequest = z.infer<typeof EditProfileRequest>;
+
+type IEditProfileResponse = z.infer<typeof EditProfileResponse>;
+
+type EditProfileResponseCode = z.infer<typeof EditProfileResponse>["code"];
+
+interface IRequest extends Omit<RequestInit, "body"> {
+  body: IEditProfileRequest;
+}
+
+const MOCK_SUCCESS: IEditProfileResponse = {
```

Comment:
Awesome that you guys are mocking up the data! I think it might not be necessary here since you are responsible for both frontend and backend!